### PR TITLE
ARROW-7344: [Packaging][Python] Build manylinux2014 wheels

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by nealrichardson** [comment]
@ursabot crossbow submit -g wheel

**Comment by nealrichardson** [comment]
@ursabot crossbow submit -g wheel

File: dev/tasks/python-wheels/azure.linux.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
         docker-compose -v
         cd arrow
         BUILD_IMAGE=centos-python-{{ wheel_tag }}
-        docker-compose pull $BUILD_IMAGE
+        docker-compose pull $BUILD_IMAGE || true
```
Comment (kou) [inline dev/tasks/python-wheels/azure.linux.yml]: We can use the `--ignore-pull-failures` option here: `docker-compose pull --ignore-pull-failures $BUILD_IMAGE`

File: docker-compose.yml [hunk 2]
```diff
@@ -590,7 +590,26 @@ services:
...
     volumes:
       - .:/arrow:delegated
-      - ./python/manylinux2010:/io:delegated
+      - ./python/manylinux201x:/io:delegated
+    command: *manylinux-command
+
+  centos-python-manylinux2014:
+    # TODO(kszucs): change to ${REPO}:tag once
+    # https://issues.apache.org/jira/browse/INFRA-19499 is resolved
+    image: arrowdev/amd64-centos-7.7-python-manylinux2014:latest
```
Comment (kou) [inline docker-compose.yml]: Could you use `${REPO}:amd64-centos-7.7-python-manylinux2014` here?

**Review by kou — CHANGES_REQUESTED** [review]
+1
We can merge this with small fixes.

We will push built image to `${REPO}:amd64-centos-7.7-python-manylinux2014`.

**Comment by nealrichardson** [comment]
Thanks @kou, had to leave a few things in there just to make sure you were paying attention ;)

I've made your suggestions, PTAL.

**Review by kszucs — APPROVED** [review]
Please add it to the [release verification script](https://github.com/apache/arrow/blob/master/dev/release/verify-release-candidate.sh#L569) before we forget it. 

Otherwise it LGTM. +1 for reusing the manylinux2010 configurations. Thanks @nealrichardson!

**Comment by nealrichardson** [comment]
@kszucs will do, also will add 3.8 to that verification (should it also have 3.5?)

**Comment by kou** [comment]
@ursabot crossbow submit -g wheel

File: dev/release/verify-release-candidate.sh [hunk 3]
```diff
@@ -555,27 +555,31 @@ check_python_imports() {
 }
 
 test_linux_wheels() {
-  local py_arches="2.7mu 3.6m 3.7m"
+  local py_arches="2.7mu 3.6m 3.7m 3.8m"
```
Comment (kou) [inline dev/release/verify-release-candidate.sh]: Could you also add `3.5m`?

Comment (kou) [reply dev/release/verify-release-candidate.sh]: `3.8` (not `3.8m`)

**Review by kou — APPROVED** [review]
+1
I've pushed built image.

We can merge this once CI is green.

**Comment by kou** [comment]
I'll merge this.

@wesm We updated release verification scripts for Linux, macOS and Windows. But they aren't tested yet because we don't have uploaded RC artifacts. You may have a problem when you use the release verification script for Windows. Sorry. We may need to fix release verification scripts on the next release process.

**Comment by wesm** [comment]
@kou no problem, I'm used to there being some problems with the verification scripts around releases, so we can fix them when it's time

**Pull request closed without merging**